moto/cloudformation/custom_model.py

```diff
@@ -0,0 +1,112 @@
+import json
+import threading
+
+from moto import settings
+from moto.core.models import CloudFormationModel
+from moto.awslambda import lambda_backends
+from uuid import uuid4
+
+
+class CustomModel(CloudFormationModel):
+    def __init__(self, region_name, request_id, logical_id, resource_name):
+        self.region_name = region_name
+        self.request_id = request_id
+        self.logical_id = logical_id
+        self.resource_name = resource_name
+        self.data = dict()
+        self._finished = False
+
+    def set_data(self, data):
+        self.data = data
+        self._finished = True
+
+    def is_created(self):
+        return self._finished
+
+    @property
+    def physical_resource_id(self):
+        return self.resource_name
+
+    @staticmethod
+    def cloudformation_type():
+        return "?"
+
+    @classmethod
+    def create_from_cloudformation_json(
+        cls, resource_name, cloudformation_json, region_name, **kwargs
+    ):
+        print("create_from_cloudformation")
```

Comment:
I noticed some print statements when running my test -- I don't know if they are intentional or whether you're still doing more work and would be cleaning up later:

```
create_from_cloudformation
{'RequestType': 'Create', 'ServiceToken': 'arn:aws:lambda:us-east-1:123456789012:function:test-cfn-lookup-ami-ids-0b', 
'ResponseURL': 'http://host.docker.internal:4615/cloudformation_us-east-1/cfnresponse?stack=arn:aws:cloudformation:us-east-1:123456789:stack/ami-lookup-failure-test/e559eb5e-f43d-4477-8791-72315d54722f', 
'StackId': 'arn:aws:cloudformation:us-east-1:123456789:stack/ami-lookup-failure-test/e559eb5e-f43d-4477-8791-72315d54722f', 
'RequestId': '8d30ba47-f16c-4e79-adef-0e43480b3d4e', 'LogicalResourceId': 'AmiIdLookup', 
'ResourceType': 'Custom::AmiIdLookup', '
ResourceProperties': {'Filters': '[{"Name":"name","Values":["amzn-ami-hvm-*"]}]', 
'Owners': '["foo"]', 'Region': 'us-east-1', 
'ServiceToken': 'arn:aws:lambda:us-east-1:123456789012:function:test-cfn-lookup-ami-ids-0b'}}
Invoking function
started
127.0.0.1 - - [31/Oct/2021 11:08:12] "POST / HTTP/1.1" 200 -
get_cfn_attribute(Id)
{}
```

Reply:
Yeah, it needs a clean-up, and more tests. Just wanted to put it out here for initial testing!